src/commands/publish.js

```diff
@@ -75,32 +76,29 @@ async function getUnpublishedPackages(packages) {
 export async function publish(opts: PublishOptions) {
   let cwd = opts.cwd || process.cwd();
   let project = await Project.init(cwd);
-  let workspaces = await project.getWorkspaces();
-  let packages = workspaces
-    .map(workspace => workspace.pkg)
-    .filter(pkg => !pkg.config.getPrivate());
+  let packages = await project.getPackages();
+  let publicPackages = packages.filter(pkg => !pkg.config.getPrivate());
   let publishedPackages = [];
 
   try {
     // TODO: Re-enable once locking issues are sorted out
     // await locks.lock(packages);
-    let unpublishedPackages = await getUnpublishedPackages(packages);
-    let isUnpublished = workspace =>
-      unpublishedPackages.some(
-        pkg => workspace.pkg.config.getName() === pkg.name
-      );
-    let unpublishedWorkspaces = workspaces.filter(isUnpublished);
-    if (unpublishedPackages.length === 0) {
+    let unpublishedPackagesInfo = await getUnpublishedPackages(packages);
```

Comment:
Nitpick, but `unpublishedPackagesInfo` doesn't seen like a great name.

Maybe `unpublishedPackages` and `unpublishedPackageNames`?

Completely not fussed though